zkevm-circuits/src/state_circuit/constraint_builder.rs

```diff
@@ -302,10 +299,13 @@ impl<F: Field> Queries<F> {
     }
 
     fn first_access(&self) -> Expression<F> {
-        not::expr(
-            self.lexicographic_ordering_upper_limb_difference_is_zero
-                .clone(),
-        ) * not::expr(self.is_storage_key_unchanged.clone())
+        or::expr(&[
```

Comment:
you're correct on everything. 

the reason that `is_storage_key_unchanged` is exists is because `lexicographic_ordering_lower_limb` covers the least significant 208 bits of the storage key and the 32 bits of the rw counter, but we need to ignore the case where the rw counter has changed but not the storage key.